csrc/type.h

```diff
@@ -766,7 +766,13 @@ enum class LoadStoreOpType {
 
 // Used to label what part of the circular buffered iterdomain
 //  a for loop is materializing.
-enum class CircularBufferLoopStage { NotApplicable, Prolog, Main, Epilog };
+enum class CircularBufferLoopStage {
+  Prolog,
+  Main,
+  Epilog,
+  NumStages,
```

Comment:
What is `NumStages`?

Reply:
This is just a placeholder for us to get the number of stages above, today it is 3 (prologue, main, epilog). In the future, there will be two more: load warp and compute warp.

This is a common trick to get the number of items in a enum, for example, in:
https://github.com/pytorch/pytorch/blob/43edb94f8a87c95762b76c2eae9753d664c9aaac/c10/core/ScalarType.h#L156
https://github.com/NVIDIA/Fuser/blob/e579cc0e84ec55d3c1e605f5d9e58825f8c8aaaf/csrc/options.h#L85

Reply:
I understand. I thought it could be related to `number_of_stages` argument for `circularBuffering`. It sounds similar in my head. https://github.com/NVIDIA/Fuser/blob/main/csrc/tensor_view.cpp#L1345

Reply:
This is a good point. Maybe we should rename it as EndOfStages